tiledb/sm/c_api/tiledb_filestore.cc

```diff
@@ -62,10 +71,10 @@ std::pair<Status, optional<std::string>> libmagic_get_mime_encoding(
     void* data, uint64_t size);
 bool libmagic_file_is_compressed(void* data, uint64_t size);
 Status read_file_header(
-    const VFS& vfs, const char* uri, std::vector<char>& header);
```

Comment:
I can't keep the const. VFS::open_file, VFS::read, and VFS::close_file are not marked const and I don't think they should be, since they internally change the state of the VFS.